surveys/views.py

```diff
@@ -31,6 +31,50 @@ class AgencyCreate(CreateView):
     success_url = '/'
 
 
+class AgencyList(ListView):
+    model = pldp_models.Agency
+    template_name = "agency_list.html"
+    context_object_name = 'agencies'
+    queryset = pldp_models.Agency.objects.all().exclude(is_active=False)
+
+
+class AgencyDetail(DetailView):
+    model = pldp_models.Agency
+    template_name = "agency_detail.html"
+    context_object_name = 'agency'
+
+    def get_context_data(self, **kwargs):
+        context = super(AgencyDetail, self).get_context_data(**kwargs)
+
+        agency = context['agency']
+
+        context['rows'] = [
+            ('Department', agency.department),
+            ('Phone', agency.phone),
+            ('Type', agency.type),
+            ('Language', agency.language.name_en),
```

Comment:
Are Agencies required to have a `language` attribute? I have an agency in my database that doesn't have a `language` so this lookup returns an error.

Reply:
Oh thanks for the catch! Yes, language is optional. Now fixed.